lib/core/util.js

```diff
@@ -192,7 +204,7 @@ function isReadableAborted (stream) {
 }
 
 function destroy (stream, err) {
-  if (stream == null || !isStream(stream) || isDestroyed(stream)) {
+  if (stream == null || (!isStream(stream) && !isH2Session(stream)) || isDestroyed(stream)) {
```

Comment:
We don't need to destroy h2 sessions?

Reply:
We need, now that the H2 session has been moved into a socket-like handling.
As the `isStream` was returning `false`, the sessions were not shutdown and the connection remain opened until timeout

Reply:
aha, then maybe the istream check is wrong, maybe `typeof obj.destroy === 'function'` is sufficient?

Reply:
I though sessions were streams?

Reply:
Hmm, yes if we just want to destroy them; but I think `isStream` is used elsewhere and that might be more problematic 🤔 

>I though sessions were streams?

Yeah, me too; had hard time guessing why was not being picked 😅